.github/workflows/update.yml

```diff
@@ -0,0 +1,25 @@
+name: update-dependencies
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened] # kittu remove this in final PR
+  schedule:
+    - cron: '0 4 * * 6'  # Every Saturday at 4 AM US time (America/New_York)
+
+jobs:
+  update-refs:
+    name: Update ref for ${{ matrix.repo.repository }}
+    strategy:
+      matrix:
+        repo:
+          - {repository: "duckdb/duckdb", env_var: "DUCKDB_REF", ref_type: "release"}
+          - {repository: "duckdb/duckdb-aws", env_var: "DUCKDB_AWS_REF", ref_type: "tip"}
+          - {repository: "duckdb/duckdb-httpfs", env_var: "DUCKDB_HTTPFS_REF", ref_type: "tip"}
+          - {repository: "substrait-io/duckdb-substrait-extension", env_var: "DUCKDB_SUBSTRAIT_REF", ref_type: "tip"}
+          - {repository: "duckdb/duckdb-iceberg", env_var: "DUCKDB_ICEBERG_REF", ref_type: "tip"}
+    uses: ./.github/workflows/deps.yaml
```

Comment:
I don't think this should run `deps.yaml` it should create a PR with any changes: which will trigger `deps.yaml` and allow us to check, update, modify the created PR